tests/src/union_compatibility_test.rs

```diff
@@ -0,0 +1,61 @@
+#[cfg(test)]
+mod tests {
+    use molecule::prelude::*;
+
+    static UNION_FOO_0_7_3_JSON_INTERMEDIATE: &str =
+        include_str!(concat!(env!("OUT_DIR"), "/0_7_3/union_foo_0_7_3.json"));
+
+    #[test]
+    fn test_recover_0_7_3_intermediate_by_current_ir_recover() {
+        let format = codegen_dev::IntermediateFormat::JSON;
+        let ast_result = format.recover(UNION_FOO_0_7_3_JSON_INTERMEDIATE.as_bytes());
+        assert!(ast_result.is_ok());
+    }
+
```

Comment:
@quake This unit test's behavior is: 

Load an `IR` result which generated by a `v0.7.3` molecule, and deserialize it by the newer `IR` format which modified by this PR.